src/main/java/com/esri/arcgisruntime/toolkit/FloorFilter.java

```diff
@@ -0,0 +1,715 @@
+    }
+    return sites;
+  }
+
+  /**
+   * Gets all the FloorFacilities associated with the FloorManager.
+   *
+   * @return an observable list of FloorFacilities, empty if FloorManager is null.
+   * @since 100.14.0
+   */
+  public ObservableList<FloorFacility> getFacilities() {
+    if (getFloorManager() != null) {
+      facilities.setAll(getFloorManager().getFacilities());
+    } else {
+      facilities.clear();
+    }
+    return facilities;
+  }
+
+  /**
+   * Gets all the FloorLevels associated with the FloorManager.
+   *
+   * @return an observable list of FloorLevels, empty if FloorManager is null.
+   * @since 100.14.0
+   */
+  public ObservableList<FloorLevel> getLevels() {
+    if (getFloorManager() != null) {
+      levels.setAll(getFloorManager().getLevels());
+    } else {
+      levels.clear();
+    }
+    return levels;
+  }
+
+  /**
+   * Gets the selected FloorSite.
+   *
+   * @return the selected floor site
+   * @since 100.14.0
+   */
+  public FloorSite getSelectedSite() {
+    return selectedSiteProperty.get();
+  }
+
+  /**
+   * Returns a property for the selected FloorSite.
+   *
+   * @return a property for the selected FloorSite
+   * @since 100.14.0
+   */
+  public SimpleObjectProperty<FloorSite> selectedSiteProperty() {
+    return selectedSiteProperty;
+  }
+
+  /**
+   * Sets the selected FloorSite if it is not already selected.
+   * In addition, the method resets the currently selected Facility to keep the data in-sync and updates the viewpoint
+   * of the GeoView to the extent of the selected FloorSite's geometry.
+   *
+   * @param newValue the selected floor site.
+   * @since 100.14.0
+   */
+  public void setSelectedSite(FloorSite newValue) {
+    selectedSiteProperty.set(newValue);
+  }
+
+  /**
+   * Returns the ID of the selected FloorSite.
+   *
+   * @return the ID of the selected FloorSite
+   * @since 100.14.0
+   */
+  public String getSelectedSiteId() {
+    var selectedSite = getSelectedSite();
+    String id = null;
+    if (selectedSite != null) {
+      id = selectedSite.getSiteId();
+    }
+    return id;
+  }
+
+  /**
+   * Sets the selected FloorSite by ID.
+   *
+   * @param id the ID of the FloorSite to select
+   * @throws NullPointerException if id is null
+   * @since 100.14.0
+   */
+  public void setSelectedSiteById(String id) {
+    Objects.requireNonNull(id);
+    getFloorManager().getSites().stream().filter(site -> id.equals(site.getSiteId())).findFirst()
+      .ifPresentOrElse(this::setSelectedSite, () -> setSelectedSite(null));
+  }
+
+  /**
+   * Handles changes to the selected site property by keeping other data in-sync.
+   *
+   * @param oldValue the previously selected site
+   * @param newValue the selected site
+   * @since 100.14.0
+   */
+  public void handleUpdateSelectedSite(FloorSite oldValue, FloorSite newValue) {
+    if (newValue != oldValue) {
+      try {
+        if (newValue != null && !blockViewpointUpdate && !siteSetViaFacility) {
```

Comment:
Do you need to check `!siteSetViaFacility` here? Since you are only setting a viewpoint it feels like `!blockViewpointUpdate` is all you might need to check.

Reply:
Thanks I've refactored this to make it less confused. I have re-used the `blockViewpointUpdate` variable and set it to true when setting the site via the facility, as I am needing to block the viewpoint being re-set to the site after a facility has been selected.